### DISCUSSION TITLE
Vault Backend - Namespace support

### DISCUSSION THREAD

**Discussion by RossKohler** [apache/airflow#44814] [Q&A]
Hi,

In the documentation for configuring Vault as a secrets backend [here](https://airflow.apache.org/docs/apache-airflow-providers-hashicorp/stable/secrets-backends/hashicorp-vault.html) it's mentioned that all the ENV variables supported by Vault [here](https://developer.hashicorp.com/vault/docs/commands#environment-variables) are also supported by the operator. I'm trying to interface with Vault Enterprise which has a concept of 'namespaces' so when I try and retrieve connections and variables I need the client to look within a specific namespace i.e. 'dev'. I'm finding that the backend operator is not respecting my `VAULT_NAMESPACE' env variable which I'm setting and thus the request the client is making is not using the correct URL. The URL is missing a namespace component. Is there any way I can get this to work?

In HVAC I have validated that my configuration works if the namespace is explicitly passed to the client.

I'm using Airflow 2.10.3.

Thanks.

**Comment by potiuk** [comment]
My best advice would be to take a look at the code -there is no indication of namespace parameter being supported, but maybe it's as easy as adding a kwarg parameter to configuration? Or maybe small modification to the underlying code could fix it - then one way you could approach it is to copy the Secrets Backend code (and name it "MyVaultSecretsBackend", configure it to be used in your Airflow instance and  modify it so that the namespace is supported. 

And then contribute it back so that you can replace it with future backend when new version is released.

**Comment by anorum-lz** [comment]
@RossKohler , To use the Vault backend with a specific namespace in Vault, you can include the namespace in the `AIRFLOW__SECRETS__BACKEND_KWARGS`. Here’s an example from one of our deployments:
```
{
  "url": "vault",
  "auth_type": "kubernetes",
  "auth_mount_point": "data-eng-kubernetes",
  "kubernetes_role": "airflow",
  "mount_point": "airflow",
  "connections_path": "",
  "variables_path": "",
  "namespace": "data-eng"
}
```

This points all the vault requests to our `data-eng` vault namespace. Hope this helps!
